core/src/main/java/org/apache/iceberg/util/ZOrderByteUtils.java

```diff
@@ -76,19 +76,15 @@ public static ByteBuffer longToOrderedBytes(long val, ByteBuffer reuse) {
    * ByteBuffer)}
    */
   public static ByteBuffer shortToOrderedBytes(short val, ByteBuffer reuse) {
-    ByteBuffer bytes = ByteBuffers.reuse(reuse, PRIMITIVE_BUFFER_SIZE);
```

Comment:
a) should we replace contents of `intToOrderedBytes` to call `longToOrderedBytes` ?
b) Replace all other `intToOrderedBytes` to call `longToOrderedBytes` to avoid double casting (type -> int -> long)

Reply:
let me rephrase this comment. 

a) Why didn't we handle the same for `intToOrderedBytes`? It is a duplicate of `longToOrderedBytes`
b) Why are we not calling `longToOrderedBytes` in the new places? calling `intToOrderedBytes` will leads to double casting.